### DISCUSSION TITLE
Import data from Way of Life

### DISCUSSION THREAD

**Discussion by VijayS1** [iSoron/uhabits#284] [Feature Requests]
How can I import data from Way of Life App? 
PFA sample csv. rename extension if needed
[Way of Life (6-4-17).txt](https://github.com/iSoron/uhabits/files/969584/Way.of.Life.6-4-17.txt)

**Comment by iSoron** [comment]
Hi @VijayS1, currently it is not possible to import data from this app, but this would be an interesting feature to have. Thank you for providing a sample CSV.

**Comment by VijayS1** [comment]
Thanks for taking a look at it. Anyway for me to convert this into existing formats that loop does accept?

**Comment by olejorgenb** [comment]
If would be very useful if the app supported import from an generic csv format. That way users could quite easily write their own import scripts. It's a lot easier to generate what basically amounts to a list of dates (in the simplest case) than to convert to the more complex formats already supported for import.

I have some logs (datestamped lines basically) that would be cool to visualize with the app. With a generic import format conversion would be very easy. 

Just saw #5. No reason is give for the wontfix tag. Why not have an accompanying "import" entry in the habits detail menu? Should accept the same format that get exported when a single habit is exported. Simply a list of dates for a single habit, no?

I see that the exported zip file contains a bit more (two almost identical checkmarks.cvs files for instance?), but only the checkmarks.csv is relevant for import I guess.

**Reply by JOduMonT** [reply]
I have interest into importing a set of task in, like it's possible in Habitatica.
I think that could give a way to the community to share their tasks and inspire others.
@iSoron Does it still in your plan to support CSV import?

**Reply by iSoron** [reply]
@JOduMonT CSV import would still be nice to have. Pull requests are very welcome.

**Comment by olejorgenb** [comment]
I see now that the Habitbull csv format is very simple.

    HabitName,HabitDescription,HabitCategory,CalendarDate,Value,CommentText
    Workout,Did you workout today?,,2017-09-23,1
    Workout,Did you workout today?,,2017-09-28,1
    ...

Ref: #44, https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/main/java/org/isoron/uhabits/core/io/HabitBullCSVImporter.java

NB: haven't tested if values containing space need to be quoted or not

**Reply by JOduMonT** [reply]
Did you try to add habit in a HabitBull CSV format to import it ?
